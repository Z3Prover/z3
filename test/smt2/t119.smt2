
; Copyright (c) 2015 Microsoft Corporation



(declare-const a Real)
(declare-const b Real)
(declare-const c Real)

(set-option :produce-models true)

(assert (= (+ (* 2.0 a) (* 3.0 b) (* 2.0 c)) 10.0))
(assert (>= c a))
(assert (>= c b))

(apply (and-then simplify solve-eqs) :print-model-converter false)

(assert (= (+ c b) 0.0))

(apply (and-then simplify solve-eqs) :print-model-converter false)

(apply (and-then simplify (! solve-eqs :theory-solver false)) :print-model-converter false)

(reset)

(declare-const a Int)
(declare-const b Int)
(declare-const c Int)

(set-option :produce-models true)

(assert (= (+ (* 2 a) (* 3 b) (* 2 c)) 10))
(assert (>= c a))
(assert (>= c b))

(apply (and-then simplify solve-eqs) :print-model-converter false)

(assert (= (+ c b a) 1))

(apply (and-then simplify solve-eqs) :print-model-converter false)

(reset)

(declare-const a Int)
(declare-const b Int)
(declare-fun f (Int) Int)

(set-option :produce-models true)

(assert (= (+ a (f a)) 20))

(apply (and-then simplify solve-eqs) :print-model-converter false)

(assert (= (+ a (f a) b) 20))
(assert (>= b 10))
(apply (and-then simplify solve-eqs) :print-model-converter false)


