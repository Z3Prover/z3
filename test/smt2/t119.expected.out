(goals
(goal
  (>= (* 2.0 c) (+ 5.0 (* (- (/ 3.0 2.0)) b)))
  (>= c b)
  :precision precise :depth 2)
)
(goals
(goal
  (<= b (- 10.0))
  (<= b 0.0)
  :precision precise :depth 2)
)
(goals
(goal
  (= (+ (* 2.0 a) (* 3.0 b) (* 2.0 c)) 10.0)
  (>= c a)
  (>= c b)
  (= (+ c b) 0.0)
  :precision precise :depth 2)
)
(goals
(goal
  (= (+ (* 2 a) (* 3 b) (* 2 c)) 10)
  (>= c a)
  (>= c b)
  :precision precise :depth 2)
)
(goals
(goal
  (<= a (- 4))
  (<= a (- 15))
  :precision precise :depth 2)
)
(goals
(goal
  (= (+ a (f a)) 20)
  :precision precise :depth 2)
)
(goals
(goal
  (= (+ a (f a)) 20)
  (>= (+ (* (- 1) a) (* (- 1) (f a))) (- 10))
  :precision precise :depth 2)
)
